src/material/list/testing/list-harness.ts

```diff
@@ -0,0 +1,299 @@
+export class MatActionListHarness extends MatListHarnessBase<
+    typeof MatActionListItemHarness, MatActionListItemHarness, ActionListItemHarnessFilters> {
+  /** The selector for the host element of a `MatActionList` instance. */
+  static hostSelector = MAT_ACTION_LIST_SELECTOR;
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatActionListHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which action list instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: ActionListHarnessFilters = {}): HarnessPredicate<MatActionListHarness> {
+    return new HarnessPredicate(MatActionListHarness, options);
+  }
+
+  _itemHarness = MatActionListItemHarness;
+}
+
+/** Harness for interacting with a standard mat-nav-list in tests. */
+export class MatNavListHarness extends MatListHarnessBase<
+    typeof MatNavListItemHarness, MatNavListItemHarness, NavListItemHarnessFilters> {
+  /** The selector for the host element of a `MatNavList` instance. */
+  static hostSelector = MAT_NAV_LIST_SELECTOR;
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatNavListHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which nav list instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: NavListHarnessFilters = {}): HarnessPredicate<MatNavListHarness> {
+    return new HarnessPredicate(MatNavListHarness, options);
+  }
+
+  _itemHarness = MatNavListItemHarness;
+}
+
+/** Harness for interacting with a standard mat-selection-list in tests. */
+export class MatSelectionListHarness extends MatListHarnessBase<
+    typeof MatListOptionHarness, MatListOptionHarness, ListOptionHarnessFilters> {
+  /** The selector for the host element of a `MatSelectionList` instance. */
+  static hostSelector = 'mat-selection-list';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatSelectionListHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which selection list instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: SelectionListHarnessFilters = {}):
+      HarnessPredicate<MatSelectionListHarness> {
+    return new HarnessPredicate(MatSelectionListHarness, options);
+  }
+
+  _itemHarness = MatListOptionHarness;
+
+  /** Whether the selection list is disabled. */
+  async isDisabled(): Promise<boolean> {
+    return await (await this.host()).getAttribute('aria-disabled') === 'true';
+  }
+
+  /**
+   * Selects all items matching any of the given filters.
+   * @param filters Filters that specify which items should be selected.
+   */
+  async checkItems(...filters: ListOptionHarnessFilters[]): Promise<void> {
```

Comment:
`select` and `deselect` instead of `check`/`uncheck`?

Reply:
For some reason I thought we had used `check`/`uncheck` for other `<option>` based components, but it looks like I was wrong. Maybe I was thinking about `MatRadio`